py3status/modules/speedtest.py

```diff
@@ -0,0 +1,298 @@
+    {upload}                upload rate to speedtest server, human readable
+    {upload_unit}           unit used for upload, eg 'MB/s'
+
+
+Color thresholds:
+    xxx: print a color based on the value of `xxx` placeholder
+
+The module will be triggered on click only. Not at start.
+
+Note that all placeholders have a version prefixed by `previous_`, eg `previous_download`.
+This can be usefull to compare two runs.
+
+Requires:
+    speedtest-cli: Command line interface for testing internet bandwidth using speedtest.net
+
+@author Cyril Levis (@cyrinux)
+
+Examples:
+```
+# colored based on thresholds
+speedtest {
+    thresholds = {
+        "download": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+        "ping": [(200, "bad"), (150, "orange"), (100, "degraded"), (10, "good")],
+        "upload": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+    }
+    format = '[\?color=ping {ping} ms] [\?color=upload Up {upload} {upload_unit}]'
+    format += ' [\?color=download Down {download} {download_unit}]'
+}
+
+# colored based on comparation between two last runs
+speedtest {
+    format = '[[\?if=download<previous_download&color=degraded Faster]'
+    format += '|[\?color=good Slower]]'
+}
+
+# compare only download
+speedtest {
+    format = '[[\?if=download<previous_download ↓ {download} {download_unit}]'
+    format += '[\?if=download<previous_download ↑ {download} {download_unit}]'
+    format += '[\?if=download=previous_download -> {download} {download_unit}]'
+    format += '|speedtest'
+}
+```
+
+SAMPLE OUTPUT
+[
+    {"full_text": "Speedtest " },
+    {"full_text": "ping ", "color": "#a9a9a9"},
+    {"full_text": "9.498 ms", "color": "#ebdbb2"},
+    {"full_text": " up ", "color": "#a9a9a9"},
+    {"full_text": "78.42 MB/s", "color": "#fb4934"},
+    {"full_text": " down ", "color": "#a9a9a9"},
+    {"full_text": "39.58 MB/s", "color": "#fb4934"}
+]
+"""
+
+from json import loads
+import time
+import threading
+
+STRING_NOT_INSTALLED = "not installed"
+
+
+class SpeedtestCli(threading.Thread):
+    """
+    A thread running speedtest-cli
+    """
+
+    def __init__(self, parent):
+        super(SpeedtestCli, self).__init__()
+        self.parent = parent
+
+    def run(self):
+        self.parent._start()
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_share = None
+    format = (
+        u"Speedtest[\?color=darkgray  {elapsed_time} s ]"
+        "[\?color=darkgray  ping [\?color=ping {ping} ms ]"
+        "up [\?color=upload {upload} {upload_unit}] "
+        "down [\?color=download {download} {download_unit}]]"
+    )
+    server_id = None
+    si_units = False
+    thresholds = {
+        "download": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+        "ping": [(200, "bad"), (150, "orange"), (100, "degraded"), (10, "good")],
+        "upload": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+    }
+    timeout = 10
```

Comment:
`timeout` should be `None` so it does not get added to the command too in case speedtest-cli changes its default timeout in the future. (imho)